### PR TITLE
Customize admin grids

### PR DESCRIPTION
**Pull request by next-sentence**

### CODE REVIEW

File: src/Model/Callout.php [hunk 1]
```diff
@@ -58,6 +58,11 @@ public function __construct()
         $this->initializeTranslationsCollection();
     }
 
+    public function __toString()
```
Comment (Roshyo) [inline src/Model/Callout.php:61]: Missing return type typing

File: src/Model/CalloutRule.php [hunk 2]
```diff
@@ -18,6 +18,11 @@ class CalloutRule implements CalloutRuleInterface
     /** @var array */
     protected $configuration = [];
 
+    public function __toString()
```
Comment (Roshyo) [inline src/Model/CalloutRule.php:21]: Missing return type typing

File: src/Resources/views/Admin/Grid/Field/callouts.html.twig [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<ul>
+{% for callout in data|default([]) %}
+    <li>
+        {{ callout }}
```
Comment (Roshyo) [inline src/Resources/views/Admin/Grid/Field/callouts.html.twig:4]: Can't we just use `callout.text` instead ? as in front.

**Review by Roshyo — CHANGES_REQUESTED** [review]

**Review by Roshyo — APPROVED** [review]

File: src/Message/Handler/AssignEligibleCalloutsToProductHandler.php [hunk 4]
```diff
@@ -19,17 +18,12 @@ final class AssignEligibleCalloutsToProductHandler implements MessageHandlerInte
     /** @var ProductRepositoryInterface */
     private $productRepository;
 
-    /** @var ObjectManager */
-    private $productManager;
-
     public function __construct(
         EligibleCalloutsProviderInterface $eligibleCalloutsProvider,
-        ProductRepositoryInterface $productRepository,
-        ObjectManager $productManager
```
Comment (loevgaard) [inline src/Message/Handler/AssignEligibleCalloutsToProductHandler.php:28]: This change essentially makes this is a breaking change, but I guess that's okay

File: src/Resources/config/app/grids.yaml [hunk 5]
```diff
@@ -5,3 +5,4 @@ sylius_grid:
     templates:
         action:
             setono_callout_default: "@SetonoSyliusCalloutPlugin/Admin/Grid/Action/setono_callout_default.html.twig"
+            callouts_product_list: "@SetonoSyliusCalloutPlugin/Admin/Grid/Action/calloutsProductList.html.twig"
```
Comment (loevgaard) [inline src/Resources/config/app/grids.yaml:8]: Should be `setono_callouts_product_list` for consistency

File: src/Message/Handler/AssignEligibleCalloutsToProductHandler.php (hunk 4 above)
Comment (vvasiloi) [reply src/Message/Handler/AssignEligibleCalloutsToProductHandler.php:28]: It's not yet stable, so it's ok.

**Comment by vvasiloi** [comment]
@loevgaard Don't release yet. I will do a follow-up PR. Let me know if there's more things to change before I start.

File: src/Resources/config/grids/setono_sylius_callout_admin_callout.yaml [hunk 6]
```diff
@@ -50,5 +71,8 @@ sylius_grid:
                 item:
                     update:
                         type: update
+                    callouts_product_list:
+                        type: callouts_product_list
```
Comment (igormukhingmailcom) [inline src/Resources/config/grids/setono_sylius_callout_admin_callout.yaml:75]: `setono_callouts_product_list` too

**Pull request merged**